### PR TITLE
feat(chat): add focus on new created folder (Issue #1589, #1512)

### PR DESCRIPTION
**Pull request by Derikyan**
**Description:**

With current implementation user need to select new folder after it was created, it is not selected automatically.
Focus should be set to new folder when user created and saved name.

Additionally fixes:
- For a newly created folder, a request is no longer sent during a click,
- When the set name is repeated, the default name is automatically set

Issues:

- Issue #1589 

**UI changes**

Before:
Снимок экрана 2024-06-27 в 09 42 37
Снимок экрана 2024-06-27 в 09 42 30

After:
Снимок экрана 2024-06-27 в 09 43 05
Снимок экрана 2024-06-27 в 09 43 14

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1674.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1674.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1674.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1674.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1674.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Chat/ChangePathDialog.tsx [hunk 1]
```diff
@@ -152,6 +174,18 @@ export const ChangePathDialog = ({
...
+      const folderName = getNextDefaultName(
+        t(DEFAULT_FOLDER_NAME),
+        folders,
+        0,
+        false,
+        true,
+      );
+
+      setSelectedFolderId(
+        constructPath(parentFolderId || getConversationRootId(), folderName),
```
Comment (denys-kolomiitsev) [inline apps/chat/src/components/Chat/ChangePathDialog.tsx]: Should it be just a `rootFolderId` instead of `getConversationRootId()`?

Comment (Derikyan) [reply apps/chat/src/components/Chat/ChangePathDialog.tsx]: sure, i'll fix it

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1674.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1674.nightly-test.deltixhub.io
>E2E tests status: success

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**